ostd/src/arch/loongarch/trap/trap.rs

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::arch::{asm, global_asm};
+
+use crate::arch::cpu::context::GeneralRegs;
+
+#[cfg(target_arch = "loongarch64")]
+global_asm!(
+    r"
+    .equ XLENB, 8
+    .macro LOAD_SP a1, a2
+        ld.d \a1, $sp, \a2*XLENB
+    .endm
+    .macro STORE_SP a1, a2
+        st.d \a1, $sp, \a2*XLENB
+    .endm
+"
+);
+
+global_asm!(include_str!("trap.S"));
+
+/// Initialize exception and interrupt handling for the current core.
+///
+/// # Safety
+///
+/// This function will:
+/// - Set `eentry` to `trap_entry`
+///
+/// You **MUST NOT** modify these registers later.
+pub unsafe fn init() {
+    // When VS=0, the entry address for all exceptions and interrupts is the same
+    loongArch64::register::ecfg::set_vs(0);
+    // Configure the entry address for normal exceptions and interrupts
+    loongArch64::register::eentry::set_eentry(trap_entry as usize);
+}
+
+/// Trap frame of kernel interrupt
+///
+/// # Trap handler
+///
+/// You need to define a handler function like this:
+///
+/// ```no_run
+/// #[no_mangle]
+/// pub extern "C" fn trap_handler(tf: &mut TrapFrame) {
+///     println!("TRAP! tf: {:#x?}", tf);
+/// }
+/// ```
+#[derive(Debug, Default, Clone, Copy)]
+#[repr(C)]
+pub struct TrapFrame {
+    /// General registers
+    pub general: GeneralRegs,
+    /// Pre-exception Mode Information
+    pub prmd: usize,
+    /// Exception Return Address
+    pub era: usize,
+    /// Extended Unit Enable
+    pub euen: usize,
+}
+
+/// Saved registers on a trap.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub(in crate::arch) struct RawUserContext {
+    /// General registers
+    pub(in crate::arch) general: GeneralRegs,
+    /// Pre-exception Mode Information
+    pub(in crate::arch) prmd: usize,
+    /// Exception Return Address
+    pub(in crate::arch) era: usize,
+    /// Extended Unit Enable
+    pub(in crate::arch) euen: usize,
+}
+
+impl Default for RawUserContext {
+    fn default() -> Self {
+        Self {
+            general: GeneralRegs::default(),
+            prmd: 0b111, // User mode, enable interrupt
+            era: 0,
+            euen: 0,
+        }
+    }
+}
+
+impl RawUserContext {
+    /// Go to user space with the context, and come back when a trap occurs.
```

Comment:
Some cleanup comments have already been made in 0fce977b4087c989de25e12e3516488cb4d30442. Could you use similar methods to achieve the same result?

Example (you should find more in the commit):
```patch
-    /// Go to user space with the context, and come back when a trap occurs.
+    /// Goes to user space with the context, and comes back when a trap occurs.
```
```patch
-/// Cpu context, including both general-purpose registers and FPU state.
+/// Userspace CPU context, including both general-purpose registers and FPU state.
```